register/src/register.rs

```diff
@@ -0,0 +1,366 @@
+use clap::{
+    crate_description, crate_name, crate_version, App, AppSettings, Arg, ArgMatches, SubCommand,
+};
+use serde_derive::Serialize;
+use solana_client::rpc_client::RpcClient;
+use solana_config_api::{config_instruction, ConfigState};
+use solana_sdk::message::Message;
+use solana_sdk::signature::{read_keypair, Keypair, KeypairUtil};
+use solana_sdk::transaction::Transaction;
+use std::error;
+use std::process::exit;
+
+pub const MAX_SHORT_FIELD_LENGTH: usize = 70;
+pub const MAX_LONG_FIELD_LENGTH: usize = 256;
+pub const JSON_RPC_URL: &str = "https://api.testnet.solana.com/";
+
+// Config account key: Va1idator1nfo111111111111111111111111111111
+pub const REGISTER_CONFIG_KEY: [u8; 32] = [
+    7, 81, 151, 1, 116, 72, 242, 172, 93, 194, 60, 158, 188, 122, 199, 140, 10, 39, 37, 122, 198,
+    20, 69, 141, 224, 164, 241, 111, 128, 0, 0, 0,
+];
+
+solana_sdk::solana_name_id!(
+    REGISTER_CONFIG_KEY,
+    "Va1idator1nfo111111111111111111111111111111"
+);
+
+#[derive(Debug, Serialize)]
+struct ValidatorInfo {
+    #[serde(with = "key_value_vec")]
+    info: Vec<String>,
```

Comment:
does this form make it easy for clients?  would a simple string suffice?

Reply:
Not easier, but I think not harder. And it takes up less space.

Reply:
Will go that direction instead.

Reply:
msgpack is nice: https://crates.io/crates/msgpack